From 0d6441bcb01db889e2e9c6b0a1244ce45725a345 Mon Sep 17 00:00:00 2001
From: fanta <tobiasmuellner@outlook.com>
Date: Tue, 25 Jun 2024 11:54:44 +0200
Subject: [PATCH] don't cache Structure BoundingBoxes

---
 .../level/levelgen/structure/StructureStart.java      | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/main/java/net/minecraft/world/level/levelgen/structure/StructureStart.java b/src/main/java/net/minecraft/world/level/levelgen/structure/StructureStart.java
index c92a2c5bb..a734a0de3 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/structure/StructureStart.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/structure/StructureStart.java
@@ -29,8 +29,6 @@ public final class StructureStart {
     private final PiecesContainer pieceContainer;
     private final ChunkPos chunkPos;
     private int references;
-    @Nullable
-    private volatile BoundingBox cachedBoundingBox;
 
     // CraftBukkit start
     private static final org.bukkit.craftbukkit.persistence.CraftPersistentDataTypeRegistry DATA_TYPE_REGISTRY = new org.bukkit.craftbukkit.persistence.CraftPersistentDataTypeRegistry();
@@ -80,14 +78,7 @@ public final class StructureStart {
     }
 
     public BoundingBox getBoundingBox() {
-        BoundingBox structureboundingbox = this.cachedBoundingBox;
-
-        if (structureboundingbox == null) {
-            structureboundingbox = this.structure.adjustBoundingBox(this.pieceContainer.calculateBoundingBox());
-            this.cachedBoundingBox = structureboundingbox;
-        }
-
-        return structureboundingbox;
+        return this.structure.adjustBoundingBox(this.pieceContainer.calculateBoundingBox());
     }
 
     public void placeInChunk(WorldGenLevel world, StructureManager structureAccessor, ChunkGenerator chunkGenerator, RandomSource random, BoundingBox chunkBox, ChunkPos chunkPos) {
-- 
2.44.0.windows.1

