From 616304f12d45961d3871adb09fc016a20e89a47f Mon Sep 17 00:00:00 2001
From: Brokkonaut <hannos17@gmx.de>
Date: Fri, 29 Jul 2022 00:38:15 +0200
Subject: [PATCH] Do not add unused blockstate coordinates

---
 .../bukkit/craftbukkit/inventory/CraftMetaBlockState.java    | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
index 601cce5f1..9d5aad794 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
@@ -281,6 +281,11 @@ public class CraftMetaBlockState extends CraftMetaItem implements BlockStateMeta
         Preconditions.checkArgument(blockStateType == blockState.getClass() && blockState instanceof CraftBlockEntityState, "Invalid blockState for " + this.material);
 
         this.blockEntityTag = ((CraftBlockEntityState) blockState).getSnapshotNBT();
+        // Brokko start - remove coordinates
+        this.blockEntityTag.remove("x");
+        this.blockEntityTag.remove("y");
+        this.blockEntityTag.remove("z");
+        // Brokko end
         // Set shield base
         if (this.material == Material.SHIELD) {
             this.blockEntityTag.putInt(CraftMetaBanner.BASE.NBT, ((CraftBanner) blockState).getBaseColor().getWoolData());
-- 
2.32.0.windows.1

