From 63d34494299909e7eb618b7dce7024f6b85de446 Mon Sep 17 00:00:00 2001
From: Brokkonaut <hannos17@gmx.de>
Date: Sun, 20 Apr 2025 04:37:12 +0200
Subject: [PATCH] Always call PlayerMoveEvent when a player moves

---
 .../server/network/ServerGamePacketListenerImpl.java      | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index f49a2c1..5d0e1cd 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -614,10 +614,12 @@ public class ServerGamePacketListenerImpl
                 Location to = CraftLocation.toBukkit(packet.position(), player.getWorld(), packet.yRot(), packet.xRot());
 
                 // Prevent 40 event-calls for less than a single pixel of movement >.>
-                double delta = Mth.square(this.lastPosX - to.getX()) + Mth.square(this.lastPosY - to.getY()) + Mth.square(this.lastPosZ - to.getZ());
-                float deltaAngle = Math.abs(this.lastYaw - to.getYaw()) + Math.abs(this.lastPitch - to.getPitch());
+                // Brokko start - Always call move event
+                // double delta = Mth.square(this.lastPosX - to.getX()) + Mth.square(this.lastPosY - to.getY()) + Mth.square(this.lastPosZ - to.getZ());
+                // float deltaAngle = Math.abs(this.lastYaw - to.getYaw()) + Math.abs(this.lastPitch - to.getPitch());
 
-                if ((delta > 1f / 256 || deltaAngle > 10f) && !this.player.isImmobile()) {
+                if (this.lastPosX != to.getX() || this.lastPosY != to.getY() || this.lastPosZ != to.getZ() || this.lastYaw != to.getYaw() || this.lastPitch != to.getPitch()) {
+                    // Brokko end - Always call move event
                     this.lastPosX = to.getX();
                     this.lastPosY = to.getY();
                     this.lastPosZ = to.getZ();
-- 
2.32.0.windows.1

