From 1fbb7b531fdd5915438670079f9a1b5d66c61ecd Mon Sep 17 00:00:00 2001
From: Brokkonaut <hannos17@gmx.de>
Date: Wed, 11 Jun 2025 06:36:36 +0200
Subject: [PATCH] Save updated player data

---
 .../level/storage/PlayerDataStorage.java      | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/net/minecraft/world/level/storage/PlayerDataStorage.java b/net/minecraft/world/level/storage/PlayerDataStorage.java
index 4ce07e19..7e902b78 100644
--- a/net/minecraft/world/level/storage/PlayerDataStorage.java
+++ b/net/minecraft/world/level/storage/PlayerDataStorage.java
@@ -98,7 +98,24 @@ public class PlayerDataStorage {
 
         return optional.or(() -> this.load(nameAndId, ".dat_old")).map(compoundTag -> {
             int dataVersion = NbtUtils.getDataVersion(compoundTag);
-            return DataFixTypes.PLAYER.updateToCurrentVersion(this.fixerUpper, compoundTag, dataVersion);
+            compoundTag = DataFixTypes.PLAYER.updateToCurrentVersion(this.fixerUpper, compoundTag, dataVersion);
+            // Brokko start - save updated player data
+            if (dataVersion != net.minecraft.SharedConstants.getCurrentVersion().dataVersion().version()) {
+                LOGGER.info("Saving updated player data for {}", nameAndId.id().toString());
+                NbtUtils.addCurrentDataVersion(compoundTag);
+                try {
+                    Path path = getPlayerDir().toPath();
+                    Path path1 = Files.createTempFile(path, nameAndId.id().toString() + "-", ".dat");
+                    NbtIo.writeCompressed(compoundTag, path1);
+                    Path path2 = path.resolve(nameAndId.id().toString() + ".dat");
+                    Path path3 = path.resolve(nameAndId.id().toString() + ".dat_old");
+                    Util.safeReplaceFile(path2, path1, path3);
+                } catch (Exception ex) {
+                    LOGGER.warn("Failed to save player data for {}", nameAndId.id().toString(), ex);
+                }
+            }
+            // Brokko end
+            return compoundTag; // CraftBukkit - handled above
         });
     }
 
-- 
2.50.1.windows.1

