From 4ceb02d9c43cc82fa474ca1f1686c61000eb08a7 Mon Sep 17 00:00:00 2001
From: Brokkonaut <hannos17@gmx.de>
Date: Wed, 11 Jun 2025 06:36:36 +0200
Subject: [PATCH] Save updated player data

---
 .../level/storage/PlayerDataStorage.java      | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/net/minecraft/world/level/storage/PlayerDataStorage.java b/net/minecraft/world/level/storage/PlayerDataStorage.java
index c5898bfb..3f8a996e 100644
--- a/net/minecraft/world/level/storage/PlayerDataStorage.java
+++ b/net/minecraft/world/level/storage/PlayerDataStorage.java
@@ -98,7 +98,24 @@ public class PlayerDataStorage {
 
         return optTag.or(() -> this.load(nameAndId, ".dat_old")).map(tag -> {
             int version = NbtUtils.getDataVersion(tag);
-            return ca.spottedleaf.dataconverter.minecraft.MCDataConverter.convertTag(ca.spottedleaf.dataconverter.minecraft.datatypes.MCTypeRegistry.PLAYER, tag, version, ca.spottedleaf.dataconverter.minecraft.util.Version.getCurrentVersion());
+            tag = ca.spottedleaf.dataconverter.minecraft.MCDataConverter.convertTag(ca.spottedleaf.dataconverter.minecraft.datatypes.MCTypeRegistry.PLAYER, tag, version, ca.spottedleaf.dataconverter.minecraft.util.Version.getCurrentVersion());
+            // Brokko start - save updated player data
+            if (version != net.minecraft.SharedConstants.getCurrentVersion().dataVersion().version()) {
+                LOGGER.info("Saving updated player data for {}", nameAndId.id().toString());
+                NbtUtils.addCurrentDataVersion(tag);
+                try {
+                    Path path = getPlayerDir().toPath();
+                    Path path1 = Files.createTempFile(path, nameAndId.id().toString() + "-", ".dat");
+                    NbtIo.writeCompressed(tag, path1);
+                    Path path2 = path.resolve(nameAndId.id().toString() + ".dat");
+                    Path path3 = path.resolve(nameAndId.id().toString() + ".dat_old");
+                    Util.safeReplaceFile(path2, path1, path3);
+                } catch (Exception ex) {
+                    LOGGER.warn("Failed to save player data for {}", nameAndId.id().toString(), ex);
+                }
+            }
+            // Brokko end
+            return tag; // CraftBukkit - handled above
         });
     }
 
-- 
2.50.1.windows.1

