From aa5cfe3b8f2086e100246a2a18d0c5273ca112bf Mon Sep 17 00:00:00 2001
From: Brokkonaut <hannos17@gmx.de>
Date: Wed, 4 Jun 2025 00:45:29 +0200
Subject: [PATCH] Do not log message, if mojang api player lookup failed

---
 .../authlib/yggdrasil/YggdrasilGameProfileRepository.java       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java b/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
index 74a2b3df..55bdacf0 100644
--- a/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
+++ b/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
@@ -122,7 +122,7 @@ public class YggdrasilGameProfileRepository implements GameProfileRepository {
             } else if (net.minecraft.server.MinecraftServer.getServer().isDebugging()) {
                 LOGGER.warn("Couldn't find profile with name: {}", name, e);
             } else {
-                LOGGER.warn("Couldn't find profile with name: {}", name);
+                // LOGGER.warn("Couldn't find profile with name: {}", name); // Brokko - no logging needed
             }
             // Paper end - only log details when debug is enabled
             return Optional.empty();
-- 
2.50.1.windows.1

