From 495508d781f4ba809746afb587c471bbcf453ec9 Mon Sep 17 00:00:00 2001
From: Brokkonaut <hannos17@gmx.de>
Date: Wed, 4 Jun 2025 00:45:29 +0200
Subject: [PATCH] Do not log message, if mojang api player lookup failed

---
 .../authlib/yggdrasil/YggdrasilGameProfileRepository.java       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java b/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
index 02717d3..367faed 100644
--- a/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
+++ b/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
@@ -115,7 +115,7 @@ public class YggdrasilGameProfileRepository implements GameProfileRepository {
         try {
             return Optional.ofNullable(client.get(HttpAuthenticationService.constantURL(nameLookupUrl + normalizeName(name)), GameProfile.class));
         } catch (final MinecraftClientException e) {
-            LOGGER.warn("Couldn't find profile with name: {}", name, e);
+            // LOGGER.warn("Couldn't find profile with name: {}", name, e); // Brokko - no logging needed
             return Optional.empty();
         }
     }
-- 
2.49.0.windows.1

