From 125b81d31e6f8b4cd2830897892f433549beb408 Mon Sep 17 00:00:00 2001
From: Brokkonaut <hannos17@gmx.de>
Date: Wed, 4 Jun 2025 00:45:29 +0200
Subject: [PATCH] Do not log message, if mojang api player lookup failed

---
 .../authlib/yggdrasil/YggdrasilGameProfileRepository.java       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java b/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
index cdab6cc..ef0ea26 100644
--- a/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
+++ b/com/mojang/authlib/yggdrasil/YggdrasilGameProfileRepository.java
@@ -116,7 +116,7 @@ public class YggdrasilGameProfileRepository implements GameProfileRepository {
         try {
             return Optional.ofNullable(client.get(HttpAuthenticationService.constantURL(nameLookupUrl + normalizeName(name)), NameAndId.class));
         } catch (final MinecraftClientException e) {
-            LOGGER.warn("Couldn't find profile with name: {}", name, e);
+            // LOGGER.warn("Couldn't find profile with name: {}", name, e); // Brokko - no logging needed
             return Optional.empty();
         }
     }
-- 
2.50.1.windows.1

